File: Examples/Node.js/Document-Properties/SetSingleDocumentProperty.js
// ExStart:1
var fs = require('fs');
var assert = require('assert');
var PdfApi = require('asposepdfcloud');
var StorageApi = require('asposestoragecloud');
var configProps = require('../Data/config.json');
var data_path = '../Data/';

var AppSID = configProps.app_sid;
var AppKey = configProps.api_key;
var outFolder = configProps.out_folder;
var config = {'appSid':AppSID,'apiKey':AppKey , 'debug' : true};

// Instantiate Aspose Storage API SDK
var storageApi = new StorageApi(config);
// Instantiate Aspose.Pdf API SDK
var pdfApi = new PdfApi(config);

// Set input file name
var name = "Sample-Annotation.pdf";
var propertyName = "AsposeDev";
var documentPropertyBody = {
		'Name' : 'AsposeDev',
		'Value' : 'Naeem Akram',
		'BuiltIn' : false
};

try {
// Upload source file to aspose cloud storage
storageApi.PutCreate(name, null, null, data_path + name , function(responseMessage) {

	assert.equal(responseMessage.status, 'OK');

	// Invoke Aspose.Pdf Cloud SDK API to set a PDF document property
	pdfApi.PutSetProperty(name, propertyName, null, null, documentPropertyBody, function(responseMessage) {
			assert.equal(responseMessage.status, 'OK');
			var docProp = responseMessage.body.DocumentProperty;
			console.log(docProp.Name + " :: " + docProp.Value);
			});
	});

}catch (e) {
  console.log("exception in example");
  console.log(e);
}

//ExEnd:1

File: Examples/Python/Examples/SetSingleDocumentProperty.py
import asposepdfcloud
from asposepdfcloud.PdfApi import PdfApi
from asposepdfcloud.PdfApi import ApiException
from asposepdfcloud.models import DocumentProperty

import asposestoragecloud
from asposestoragecloud.StorageApi import StorageApi
from asposestoragecloud.StorageApi import ResponseMessage

apiKey = "XXXXX" #sepcify App Key
appSid = "XXXXX" #sepcify App SID
apiServer = "http://api.aspose.com/v1.1"
data_folder = "../../data/"

#Instantiate Aspose Storage API SDK
storage_apiClient = asposestoragecloud.ApiClient.ApiClient(apiKey, appSid, True)
storageApi = StorageApi(storage_apiClient)
#Instantiate Aspose Pdf API SDK
api_client = asposepdfcloud.ApiClient.ApiClient(apiKey, appSid, True)
pdfApi = PdfApi(api_client);

#set input file name
name = "Sample-Annotation.pdf"
propertyName = "author"

body = DocumentProperty.DocumentProperty()
body.Name = "author"
body.Value = "Farooq Sheikh"
body.BuiltIn = True


try:
    #upload file to aspose cloud storage
    response = storageApi.PutCreate(name, data_folder + name)

    #invoke Aspose.Pdf Cloud SDK API to set a PDF document property
    response = pdfApi.PutSetProperty(name, propertyName, body)

    if response.Status == "OK":
        docProp = response.DocumentProperty
        print docProp.Name + " :: " + docProp.Value

except ApiException as ex:
            print "ApiException:"
            print "Code:" + str(ex.code)
            print "Message:" + ex.message
